tests/runner.py

```diff
@@ -1100,24 +1100,32 @@ def do_GET(self):
         self.end_headers()
         self.wfile.write(open(path_from_root('tests', 'browser_harness.html'), 'rb').read())
       elif 'report_' in self.path:
+        # for debugging, tests may encode the result and their own location as result|location
+        if '|' in self.path:
+          path, location = self.path.split('|', 1)
+        else:
+          path = self.path
+          location = '?'
         if DEBUG:
-          print('[server response:', self.path, ']')
+          print('[server response:', path, location, ']')
         if out_queue.empty():
-          out_queue.put(self.path)
+          out_queue.put(path)
         else:
           # a badly-behaving test may send multiple xhrs with reported results; we just care
           # about the first (if we queued the others, they might be read as responses for
           # later tests, or maybe the test sends more than one in a racy manner)
-          if DEBUG:
-            raise Exception('browser harness error, excessive response to server - test must be fixed! "%s"' % self.path)
+          # note that by placing 'None' in the queue, since if we raise an exception here it
```

Comment:
i can't quite parse this sentence

Reply:
I'll clarify in a comment,
```
          # we place 'None' in the queue here so that the outside knows something went wrong
          # (none is not a valid value otherwise; and we need the outside to know because if we
          # raise an error in here, it is just swallowed in python's webserver code - we want
          # the test to actually fail, which a webserver response can't do).
```